runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -263,10 +264,11 @@ public void onExperienceChanged(ExperienceChanged event)
 		state.setActionType(XpActionType.EXPERIENCE);
 
 		final Actor interacting = client.getLocalPlayer().getInteracting();
-		if (interacting instanceof NPC)
+		if (interacting instanceof NPC && COMBAT.contains(skill))
 		{
 			final NPC npc = (NPC) interacting;
 			xpState.updateNpcExperience(skill, npc, npcManager.getHealth(npc.getName(), npc.getCombatLevel()));
+			state.setActionType(XpActionType.ACTOR_HEALTH);
```

Comment:
Why is this line necessary? I dont think it is

Reply:
it defaults all skills to the experience type. this line was previously inside the updateNpcExperience method but I felt that it would be easier to understand what the program was doing if the actionType was set in one place (i.e. only in the onExperienceChanged method) so I refactored it to here.

Reply:
I liked it more in the updateNpcExperience